src/pushsource/_impl/reader.py

```diff
@@ -0,0 +1,23 @@
+from io import BufferedReader, SEEK_SET, UnsupportedOperation
+
+
+class PushItemReader(BufferedReader):
+    # Internal class to ensure that the file-like content object returned by
+    # the push items are read-only and non-seekable with a name attribute.
+    def __init__(self, raw, name=None, **kwargs):
+        super(PushItemReader, self).__init__(raw, **kwargs)
+
+        # Attempt to assign name from the raw object if none is provided
```

Comment:
I think that it might be better to drop this, and to just always force a `name` to come from the caller.  Every push item has a `name` attribute, so maybe that can just be used.

The way you have it now, when combined with the current `content` implementation,  means that the contract of `opener` is not only "returns a file-like object" but the more complicated "returns a file-like object with a non-empty `name` attribute".

I believe a lot of io streams do not have the `name` attribute, including those we're likely to see used in the future, such as the stream of an HTTP response body.  So, insisting that `opener` *must* return something with a `name` seems to make the API more difficult to use.